### PR TITLE
Correct dust and change

### PR DESCRIPTION
**Pull request by rnicoll**
* Correct dust threshold, which is never less than 1 DOGE.
* Correct minimum change, which is 1 DOGE.

### CODE REVIEW

File: src/primitives/transaction.h [hunk 1]
```diff
@@ -190,7 +190,8 @@ class CTxOut
             nSize += (32 + 4 + 1 + 107 + 4); // the 148 mentioned above
         }
 
-        return 3 * minRelayTxFee.GetFee(nSize);
+        // Dogecoin: Anything below 1 DOGE is always dust
+        return std::max(COIN, 3 * minRelayTxFee.GetFee(nSize));
```
Comment (langerhans) [inline src/primitives/transaction.h]: minRelayTxFee as passed from GetDogecoinDustFee() is 1 COIN, so this would return a threshold of 3*COIN 🤔

Comment (langerhans) [reply src/primitives/transaction.h]: Okay so setting this to always return COIN will indeed roll a half-COIN change into fee. This was tested with a single input tx on regtest.

Comment (rnicoll) [reply src/primitives/transaction.h]: Okay, going to have it just do that. Also; it looks like we may have a nightmare to unpick all this to enable Segwit

**Comment by ivanslav1517** [comment]
Ждите через час отвечу

Ivan+Slavynskyi+ivanslav1517@gmail.com

-----Исходное сообщение-----
От: "Max K." <notifications@github.com>
Отправлено: ‎03.‎11.‎2019 19:43
Кому: "dogecoin/dogecoin" <dogecoin@noreply.github.com>
Копия: "Subscribed" <subscribed@noreply.github.com>
Тема: Re: [dogecoin/dogecoin] Correct dust and change (#1615)

Merged #1615 into 1.14-maint.

**Comment by ivanslav1517** [comment]
Dog error stop job

Ivan+Slavynskyi+ivanslav1517@gmail.com

-----Исходное сообщение-----
От: "Max K." <notifications@github.com>
Отправлено: ‎03.‎11.‎2019 19:43
Кому: "dogecoin/dogecoin" <dogecoin@noreply.github.com>
Копия: "Subscribed" <subscribed@noreply.github.com>
Тема: Re: [dogecoin/dogecoin] Correct dust and change (#1615)

Merged #1615 into 1.14-maint.

**Pull request merged**